### ISSUE TITLE
Tooltip: AVT 1 violations

### ISSUE THREAD

**Issue by elizabethsjudd** [carbon-design-system/carbon#3100]
## Environment

> Operating system

Mac

> Browser

Chrome, Safari

> Automated testing tool and ruleset

AAT 4.2.1

> Assistive technology used to verify

VoiceOver

## Detailed description

Tooltip has multiple AAT violations

> What version of the Carbon Design System are you using?

10.3.1

> What did you expect to happen?

There to not be AAT violations

> What happened instead?

3 violations (color contrast and landmarks)

> What WCAG 2.1 checkpoint does the issue violate?

See screen shot

## Steps to reproduce the issue

1. Force the tooltip to be open by adding the `bx--tooltip--shown` class (trying to run DAP with default behaviors I couldn't get the tooltip to remain open while running the tests)
2. Run DAP

## Additional information

I would expect the dialog to have `role="dialog"` not the trigger (solve the violation about the landmark) while the trigger uses `aria-controls` attribute to create the relationship between the two elements. 

`role="tooltip"` should be be for static text only (i.e. definition tooltip #3099)

<img width="1566" alt="Screen Shot 2019-06-18 at 1 20 29 PM" src="https://user-images.githubusercontent.com/1275779/59705371-41837180-91cc-11e9-8799-4459842dbcfd.png">

**Comment by snidersd** [comment]
This issue is related to issue #2736.

**Comment by elizabethsjudd** [comment]
@snidersd FYI, I made issues for AVT 1 for all three versions of the Tooltip that I'm working through.  WH is focused on the Vanilla version of Carbon since we don't use React. I'm trying to take them in smaller chunks for contribution purposes.

**Comment by elizabethsjudd** [comment]
@carbon-design-system/design for the interactive tooltip, is it expected that mouse users have to click to see the tooltip but keyboard users only focus? Typically if an action happens on "click" for mouse users, keyboard users usually have to focus the trigger and then press `space` or `enter` to actually perform the particular action.  This is also creating some odd behavior where my focus can get trapped between the tooltip and it's trigger (if i fiddle around with the demo enough I can get the focus to release but it's not consistent).

**Comment by aagonzales** [comment]
I would expect there not to be a difference in the interaction between mouse and keyboard users. We opt for "click to open" because we're allowing for interactions inside of the tooltip and its annoying if you can hover off the tooltip area before you have a chance to trigger the interactive parts. 

Are you recommending they be different? Is it normal for behaviors to vary between mouse and keyboard? What would your ideal behavior be?

**Comment by elizabethsjudd** [comment]
@aagonzales 100% agree with having the user click on the trigger to open the tooltip as this is the best a11y experience IMO. The ideal behavior would then be for consistency for both keyboard and mouse users where they have to explicitly open the tooltip for it to display. That would mean for mouse users they would have to click (current behavior), and that keyboard users would have to focus AND press space/enter on the trigger to open the keyboard (right now they only have to focus).

**Comment by aagonzales** [comment]
@elizabethsjudd so what are the next steps? Is this issue for fixing the behavior to your recommendations or does another issue need to be open?

**Comment by elizabethsjudd** [comment]
@aagonzales I have a different issue to update the JavaScript for tooltips that is already assigned to me that was impacted by this question so as long as i have your written okay with the change (https://github.com/carbon-design-system/carbon/issues/3100#issuecomment-507843443) I'm good to actually work on it.  Thanks for getting back to me!
